### PR TITLE
SQL: fix implicit sys fields for implicit engine of partitioned table [#366]

### PR DESCRIPTION
**Pull request by kevgs**

### CODE REVIEW

File: sql/sql_table.cc [hunk 1]
```diff
@@ -8854,10 +8854,10 @@ bool mysql_alter_table(THD *thd, const char *new_db, const char *new_name,
...
   if (versioned)
   {
-    if (create_info->db_type &&
-      table->s->db_type() != create_info->db_type && (
-      table->file->native_versioned() ||
-      create_info->db_type->flags & HTON_NATIVE_SYS_VERSIONING))
+    if (create_info->db_type && table->s->db_type() != create_info->db_type &&
+        (table->file->native_versioned() ||
+         ha_check_storage_engine_flag(create_info->db_type,
+                                      HTON_NATIVE_SYS_VERSIONING)))
```
Comment (midenok) [inline sql/sql_table.cc]: Make this more readable.

**Comment by kevgs** [comment]
Fixed.

File: sql/sql_table.cc [hunk 2]
```diff
@@ -8854,13 +8854,17 @@ bool mysql_alter_table(THD *thd, const char *new_db, const char *new_name,
 
   if (versioned)
   {
-    if (create_info->db_type && table->s->db_type() != create_info->db_type &&
-        (table->file->native_versioned() ||
-         ha_check_storage_engine_flag(create_info->db_type,
-                                      HTON_NATIVE_SYS_VERSIONING)))
+    if (handlerton *hton1= create_info->db_type)
```
Comment (midenok) [inline sql/sql_table.cc]: This should bring warning about suggested parentheses.

Comment (kevgs) [reply sql/sql_table.cc]: I don't see such warning neither with gcc nor with clang.

**Review by midenok — APPROVED** [review]

File: sql/handler.cc [hunk 3]
```diff
@@ -6872,7 +6872,25 @@ bool Vers_parse_info::check_and_fix_implicit(
...
+    if (!(create_info->used_fields & HA_CREATE_USED_ENGINE) &&
+        info->partitions.elements)
+    {
+      partition_element *element=
+          static_cast<partition_element *>(info->partitions.elem(0));
+      if (handlerton *hton= element->engine_type)
+      {
+        if (ha_check_storage_engine_flag(hton, HTON_NATIVE_SYS_VERSIONING))
+          integer_fields= true;
+      }
```
Comment (midenok) [inline sql/handler.cc]: Remove `if` in `if` before merge.

**Pull request merged**